src/sidebar/components/ShareDialog/ImportAnnotations.tsx

```diff
@@ -126,13 +134,17 @@ function ImportAnnotations({
   }, [currentUser, file]);
 
   let importAnnotations;
-  if (annotations && selectedUser) {
+  if (annotations && selectedUser && userList) {
     importAnnotations = async () => {
-      const annsToImport = annotations.filter(ann => ann.user === selectedUser);
+      const userEntry = userList.find(item => item.userid === selectedUser);
+      /* istanbul ignore next - This should never be triggered */
```

Comment:
Maybe we can add a non-null assertion (`userList.find(item => item.userid === selectedUser)!`) to let TS know there has to be a result? Or do you think it's safer this way?

Reply:
I used a runtime check here because the fact that `selectedUser` _should_ match an entry in `userList` is enforced somewhere else in the file, so it is a bit easier to make a mistake and break the invariant. What I really want here is a runtime assertion to catch such a mistake if it happens:

```js
const userEntry = userList.find(...);
assert(userEntry); // Produces helpful runtime error if this fails
// `userEntry` is inferred as non-null here
```

We don't currently have good infrastructure for this though.

Reply:
Makes sense.